Controllers/RecipeController.cs

```diff
@@ -41,11 +43,12 @@ public RecipeController(AppDatabaseContext db)
     [Authorize]
     public IActionResult UpsertRecipe(RecipeModel recipe) //For upserting we need the full model information (id can be ommited for creating a new recipe)
     {
-        var email = (HttpContext.User.Identity as ClaimsIdentity)?.getEmailFromClaim();
+        recipe.Owner = (HttpContext.User.Identity as ClaimsIdentity)?.getEmailFromClaim() ?? recipe.Owner; // Set recipe owner to current user
```

Comment:
why is it changed to a completely different variable?

Reply:
This is not a new variable. The flow looks like this:
Receive recipe object from user -> Since recipe can be changed only by the user who owns it - we set the recipes owner to current user -> if this is a create operation we put the recipe into db / if this is an update operation we hard set the provided recipe email to current user and check if the model signature (id + email in this case) of db and provided model match: if they do - we update otherwise we return Bad Rquest